docs/toolkits/openstack-planning-guide.md

```diff
@@ -1,11 +1,124 @@
+| dashboard               | Horizon        | A web interface for using OpenStack     | Management Console |
+| block storage           | Cinder         | Virtual disk drives                     | EBS                |
+| object storage          | Swift          | Data storage buckets                    | S3                 |
+| filesystem              | Manila         | Virtual shared folders                  | EFS                |
+| bare metal              | Ironic         | Real, physical computers                | EC2 Bare Metal     |
+| DNS                     | Designate      | DNS host records                        | Route 53           |
+| orchestration           | Heat           | Automated control of OpenStack services | CloudFormation     |
+| container orchestration | Magnum         | Push-button Kubernetes clusters         | EKS                |
+
+
 ### Control Plane and Data Plane
 
+In a large distributed system like an OpenStack cloud, there is a logical separation between the Control Plane and Data Plane. The control plane consists of the set of OpenStack services (and their dependencies) that _manage_ users' resources on the cloud. The data plane consists of the users' resources themselves. The data plane includes virtual computers, networks, and various types of data storage.
+
+First, let's look at the control plane. It's a sub-set of the diagram at the top of this page.
+
+[![OpenStack Diagram](../img/openstack-diagram-control-plane-only.jpeg)](../img/openstack-diagram-control-plane-only.jpeg)
+
+An OpenStack control plane has a few layers. At the bottom is at least one physical computer (a.k.a. server) to run the control plane services.
+
+The next layer is infrastructure services. These are not part of OpenStack per se, but OpenStack services depend on them to deliver a functioning cloud.
+
+| Service type      | Name(s)                                         | What it does                                              |
+|-------------------|-------------------------------------------------|-----------------------------------------------------------|
+| Database          | MariaDB or MySQL (optionally in Galera cluster) | Stores the persistent state of the control plane          |
+| Message broker    | RabbitMQ                                        | Delivers messages between service components              |
+| API load balancer | HAProxy                                         | Routes and secures network requests to OpenStack services |
+
+
+
+
+
+### Load Balancing and High Availability
+
+The entire OpenStack control plane can be load-balanced and/or high-availability.
+
+When configured correctly, this means that every
+
+This means
+
+LB/HA architecture makes sense for larger clouds with downtime-intolerant workloads and hundreds of compute nodes. LB/HA is not essential when you are just getting started, and it is more complex to understand, deploy, manage, and troubleshoot.
+
+If you are 
```

Comment:
I think this is not a case of "Intentionally left blank".  Might need to add a todo here to catch it just in case.